### PR TITLE
Dynamic let

### PR DESCRIPTION
**Pull request by marcelocantos**
Fixes # .

Changes proposed in this pull request:
- Introduce `let bindings; expr` form, where `bindings` is a tuple whose attributes are dynamic names to be bound when evaluating `expr`. This allows library functions to set dynamic variables without publishing the internal names.
- 

Checklist:
- [x] Added related tests
- [ ] Made corresponding changes to the documentation

### CODE REVIEW

File: syntax/expr_let_test.go [hunk 1]
```diff
@@ -232,3 +217,33 @@ func TestExprLetGetPattern(t *testing.T) {
 
 	AssertCodeErrors(t, "", `let (x?: (k: (z?: w:42))) = (x: (y: (z: 1))); w`)
 }
+
+func TestExprLetDynIdentPattern(t *testing.T) {
+	AssertCodesEvalToSameValue(t, `42`, `let f = \x x * @{y}; let @{y} = 6; f(7)`)
```
Comment (orlade) [inline syntax/expr_let_test.go:222]: Parallel?

File: rel/expr_dynlet.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+func NewDynLetExpr(scanner parser.Scanner, bindings, expr Expr) Expr {
+	return &DynLetExpr{ExprScanner: ExprScanner{scanner}, bindings: bindings, expr: expr}
+}
+
+// String returns a string representation of the expression.
+func (e *DynLetExpr) String() string {
+	return fmt.Sprintf("(let %s; %s)", e.bindings, e.expr)
+}
+
+// Eval returns the lhs
```
Comment (orlade) [inline rel/expr_dynlet.go:29]: ?

File: syntax/expr_let_test.go [hunk 3]
```diff
@@ -232,3 +217,33 @@ func TestExprLetGetPattern(t *testing.T) {
...
+	AssertCodesEvalToSameValue(t, `42`, `let f = \x x * @{y}; let [@{y}, y] = [6, 8]; f(7)`)
+	AssertCodesEvalToSameValue(t, `42`, `let f = \x x * @{y}; let {@{y}, 8} = {6, 8}; f(7)`)
+	AssertCodesEvalToSameValue(t, `42`, `let f = \x x * @{y}; let (a:@{y}, ...) = (a:6, b:8); f(7)`)
+	AssertCodesEvalToSameValue(t, `42`, `let f = \x x * @{y}; let {1:@{y}, ...} = {1:6, 2:8}; f(7)`)
+	AssertCodesEvalToSameValue(t, `[2, 4, 8, 16, 32]`, `let f = \x x ^ @{y}; [1, 2, 3, 4, 5] >> \@{y} f(2)`)
+	AssertCodesEvalToSameValue(t, `[27, 64]`, `
```
Comment (orlade) [inline syntax/expr_let_test.go:228]: Nit: I don't know if it's just too boring or something, but these test cases would be easier to follow with simpler math, like 1 + @{y}.

Comment (marcelocantos) [reply syntax/expr_let_test.go:228]: I made it more complicated, but I'm hoping the obvious pattern makes the intent clearer.

**Review by orlade-anz — APPROVED** [review]

**Pull request merged**